### DISCUSSION TITLE
Audio to text using a microphone?

### DISCUSSION THREAD

**Discussion by Xenos71** [SubtitleEdit/subtitleedit#6232] [Ideas]
Hi,

Just recently I discovered the audio to text function, and I have to say that it does a great job with clean audio. It's a great feature, thank you for implementing it.

Would it be possible to use this function with a microphone, so as to convert voice to subtitles? 
For example, playing a video and narrating over it through a mic, and SE converting the user's voice to subtitles.

This could be very helpful in cases where the real audio is not clear (you could just repeat the lines yourself, without any explosions on the background...), and also if you want to make rough notes or comments on a video.

Thanks in advance

**Comment by niksedk** [comment]
How is this?
https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.7/SubtitleEditBeta.zip

![image](https://user-images.githubusercontent.com/1261331/190917174-a62bb016-fdcd-4650-97f3-20ca19522123.png)


It will record in the main textbox while the shortcut is pressed down (first time you will have to choose language).  Seems a bit slow and requires nice clean English... 
How does it work for you?

**Comment by Xenos71** [comment]
Hello Nikolaj,

yes, it works! First I tried it in English, but my biggest concern was how it would perform in Greek. Well, it works really good in my native language too!
It is a bit slow as you mentioned, but the important thing is that it works, and is really helpful.

One question (maybe some more will follow in the near future..):
For testing, I used my laptop which has a built-in mic, so SE automatically recognized it as the audio source for speech recognition.
If I plug in a USB mic for better quality (I don't have one right now to test it), is there a way to select it? I searched in the Options but I couldn't find something relevant.

Once again, thank you very much for your work.

Xenos






On Sunday, September 18, 2022 at 07:17:04 PM GMT+3, Nikolaj Olsson ***@***.***> wrote: 







How is this?
https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.7/SubtitleEditBeta.zip



It will record while the shortcut is pressed down... maybe,,, seems a bit slow and requires nice clean English...
How does it work for you?

**Comment by niksedk** [comment]
SE just uses the active microphone in Windows.

**Comment by Dnkhatri** [comment]
Looks like openai introduced and open sourced a Multilanguage speech to text  it will be interesting to see how well it works. 

https://openai.com/blog/whisper/

**Reply by niksedk** [reply]
I've tried to add support for Whisper in latest SE beta: https://github.com/SubtitleEdit/subtitleedit/releases/download/3.6.7/SubtitleEditBeta.zip
Should work on whole files + selected lines via context menu.

**Comment by niksedk** [comment]
I could not find the data files for whisper... can anybody make whisper work?

Also, SE now has a record icon for with recording is on (incl. volume bars)

![image](https://user-images.githubusercontent.com/1261331/191808399-4b0aa30b-3075-4745-bbdc-50ce6f373d05.png)
